coriolis_openstack_utils/resource_utils/users.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2018 Cloudbase Solutions Srl
+# All Rights Reserved.
+
+from oslo_log import log as logging
+from coriolis_openstack_utils import conf
+
+
+LOG = logging.getLogger(__name__)
+CONF = conf.CONF
+
+
+def list_users(openstack_client, filters={}):
+    return openstack_client.keystone.users.list(**filters)
+
+
+def get_user(openstack_client, user_id):
+    return openstack_client.keystone.users.get(user_id)
+
+
+def get_body(openstack_client, user_id):
+
+    relevant_keys = {'enabled', 'name', 'email'}
+    src_body = get_user(openstack_client, user_id)
+    if not isinstance(src_body, dict):
+        src_body = src_body.to_dict()
+
+    src_body = {k: v for k, v in src_body.items() if k in relevant_keys}
+
+    return src_body
+
+
+def create_user(openstack_client, body):
+    body['password'] = CONF.destination.new_user_password
+    return openstack_client.keystone.users.create(**body).id
+
+
+def add_admin_roles(openstack_client, user_name, project_name_list):
+    for project_name in project_name_list:
+        LOG.info("Adding admin role to user '%s' in project '%s'"
+                 % (user_name, project_name))
+        openstack_client.add_admin_role_to_project(project_name, user_name)
```

Comment:
Please either wrap this in a `try:-except keystone.NotFound:` and LOG if you can't find the tenant, or simply check for the tenant's existence before trying to add a role for the user in it.